sync/sync_head.go

```diff
@@ -130,6 +131,7 @@ func (s *Syncer[H]) setSubjectiveHead(ctx context.Context, netHead H) {
 	s.pending.Add(netHead)
 	s.wantSync()
 	log.Infow("new network head", "height", netHead.Height(), "hash", netHead.Hash())
+	s.metrics.observeNewSubjectiveHead(int64(netHead.Height()), netHead.Time())
```

Comment:
We should set it after a successful call to `storeHeaders.` Otherwise, it won't be updated once the node is fully synced. E.g. the store's head will go up, but the subjective head will stay still.